compiler/sem.nim

```diff
@@ -251,6 +251,8 @@ proc fixupTypeAfterEval(c: PContext, evaluated, eOrig: PNode): PNode =
       if eOrig.typ.skipTypes(abstractInst).kind == tySequence and 
          arg.typ.skipTypes(abstractInst).kind == tyArrayConstr:
         arg.typ = eOrig.typ
+  if result.info.line == -1:
```

Comment:
The code never passes through there. I placed a debug statement at the beginning and it doesn't get called.